server/observable.go

```diff
@@ -504,12 +527,50 @@ func (o *Observable) updateStore() {
 	}
 }
 
+func (o *Observable) shouldSample() bool {
```

Comment:
This called on every ack? rand might be expensive here.

Reply:
any other suggesting, I couldnt think of a faster way.

Reply:
I have been exploring wyhash. Its ok here for now, but let's comment on it so if I change it over I can remember that one too ;)

Reply:
added a todo here to remind us